front/src/applications/operationalStudies/components/MacroEditor/osrdToNge.ts

```diff
@@ -285,15 +286,37 @@ const importTimetable = async (
 
   nodes = convertGeoCoords(nodes);
 
+  const DTOLabels: NGELabel[] = [];
   // Create one NGE train run per OSRD train schedule
-  const trainruns: Trainrun[] = trainSchedules.map((trainSchedule) => ({
-    id: trainSchedule.id,
-    name: trainSchedule.train_name,
-    categoryId: DEFAULT_TRAINRUN_CATEGORY.id,
-    frequencyId: DEFAULT_TRAINRUN_FREQUENCY.id,
-    trainrunTimeCategoryId: DEFAULT_TRAINRUN_TIME_CATEGORY.id,
-    labelIds: [],
-  }));
+  let labelId = 0;
+  const trainruns: Trainrun[] = trainSchedules.map((trainSchedule) => {
+    let labelIds: number[] = [];
+    if (trainSchedule.labels) {
+      labelIds = trainSchedule.labels.map((label) => {
+        const DTOLabel = DTOLabels.find((DTOlabel) => DTOlabel.label === label);
+        if (DTOLabel) {
+          return DTOLabel.id;
+        }
+        const newDTOLabel: NGELabel = {
+          id: labelId,
+          label,
+          labelGroupId: 1,
+          labelRef: 'Trainrun',
+        };
+        DTOLabels.push(newDTOLabel);
+        labelId += 1;
```

Comment:
```suggestion
          id: labelId++,
          label,
          labelGroupId: 1,
          labelRef: 'Trainrun',
        };
        DTOLabels.push(newDTOLabel);
```

Reply:
`++` is forbidden in our style guide (and I'm not a fan of using `++` as an expression instead of a statement).

Reply:
I think eslint doesn't link  de `++` :)

Reply:
Good to know ! :)